include/fi_shm.h

```diff
@@ -101,37 +106,46 @@ struct smr_op_hdr {
 	};
 };
 
-struct smr_cmd_hdr {
-	struct smr_op_hdr	op;
-	uint64_t		msg_id;
+union smr_cmd_data {
+	uint8_t			msg[SMR_MSG_DATA_LEN];
+	struct iovec		iov[SMR_MSG_DATA_LEN / sizeof(struct iovec)];
 };
 
-#define SMR_CMD_SIZE		128	/* align with 64-byte cache line */
-#define SMR_CMD_DATA_LEN	(128 - sizeof(struct smr_cmd_hdr))
+struct smr_cmd_msg {
+	struct smr_msg_hdr	hdr;
+	union {
+		uint8_t		msg[SMR_MSG_DATA_LEN];
+		struct iovec	iov[SMR_MSG_DATA_LEN / sizeof(struct iovec)];
+	};
+};
 
-#define SMR_NAME_SIZE	32
-struct smr_addr {
-	char		name[SMR_NAME_SIZE];
-	fi_addr_t	addr;
+struct smr_rma_hdr {
+	uint64_t		rma_count;
 };
 
-#define SMR_RMA_IOV_LIMIT	3 //TODO find a way to avoid this?
-union smr_cmd_data {
-	uint8_t			msg[SMR_CMD_DATA_LEN];
-	struct iovec		iov[SMR_CMD_DATA_LEN / sizeof(struct iovec)];
-	struct ofi_rma_iov	rma_iov[SMR_CMD_DATA_LEN / sizeof(struct ofi_rma_iov)];
-	struct ofi_rma_ioc	rma_ioc[SMR_CMD_DATA_LEN / sizeof(struct ofi_rma_ioc)];
+struct smr_cmd_rma {
+	struct smr_rma_hdr	hdr;
+	struct fi_rma_iov	rma_iov[SMR_RMA_DATA_LEN /
+					sizeof(struct fi_rma_iov)];
 };
 
 struct smr_cmd {
-	struct smr_cmd_hdr	hdr;
-	union smr_cmd_data	data;
+	union {
+		struct smr_cmd_msg	msg;
+		struct smr_cmd_rma	rma;
+	};
```

Comment:
I'm assuming that we need smr_cmd defined in order to allocate some sort of queue.

Reply:
Correct